src/libs/tools/include/merging/mergingkdb.hpp

```diff
@@ -0,0 +1,111 @@
+/**
+ * @file
+ *
+ * @brief
+ *
+ * @copyright BSD License (see doc/COPYING or http://www.libelektra.org)
+ */
+
+#ifndef KDBMERGE_HPP_
+#define KDBMERGE_HPP_
+
+#include <kdb.hpp>
+#include <merging/threewaymerge.hpp>
+
+namespace kdb
+{
+
+namespace tools
+{
+
+namespace merging
+{
+
+/**
+ * Provides a merging wrapper around a KDB instance. The wrapper allows to pass
+ * a three way merger instance that is used to resolve conflicts during KDB set.
+ */
+class MergingKDB : KDB
```

Comment:
Is it a private inheritance on purpose? Isn't it a subtype? (otherwise one would write `public` there)